source/Test.cpp

```diff
@@ -74,6 +74,34 @@ namespace{
 			})
 			+ "\", or \"" + lastValidIt->second + '"';
 	}
+	
+	// Send an SDL_event to on of the UIs.
+	bool EventToUI(UI &menuOrGamePanels, const SDL_Event &event)
+	{
+		return menuOrGamePanels.Handle(event);
+	}
+	
+	// Send an keyboard input to one of the UIs.
+	bool KeyInputToUI(UI &menuOrGamePanels, const char* keyName, bool shift=false, bool ctrl=false, bool alt=false)
```

Comment:
Rather than a number of boolean flags, use a bit field, like how the game tests for shift, Ctrl, alt, etc.